### PR TITLE
Subdirs testing disabled and CI and split into build and test.

### PR DESCRIPTION
**Pull request by shyamjesal**
- Split CI into build and test.

- Misc test moved to Github actions

- Taps test removed

Signed-off-by: Shyam Jesalpura <s.jesalpura@gmail.com>

### CODE REVIEW

**Comment by ustiugov** [comment]
PR name is misleading. 

Please add the following clean-up at the end of each test in yaml that engages containerd:
```
    - name: Cleaning
      if: ${{ always() }}
      run: ./scripts/clean_fcctr.sh
```

So that if a test run fails, it would not break all further runs.

File: function-images/lr_serving/Dockerfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ RUN apk add --no-cache \
     pip install --disable-pip-version-check --no-build-isolation pandas==1.0.5 && \
     pip install --disable-pip-version-check --no-build-isolation protobuf==3.11.3 && \
     pip install --disable-pip-version-check --no-build-isolation grpcio==1.26.0 && \
+    pip install --disable-pip-version-check --no-build-isolation pybind11 && \
```
Comment (ustiugov) [inline function-images/lr_serving/Dockerfile:16]: can you please explain the fix?

File: function-images/pyaes/Dockerfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ RUN pip3 install --user -r requirements.txt
 
 FROM vhiveease/py_grpc:base as var_workload
 COPY *.py /
+RUN true
```
Comment (ustiugov) [inline function-images/pyaes/Dockerfile:8]: can you please explain the fix here and elsewhere?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    - name: Build
+      run: go build -race -v -a ./...
+
+    - name: Run tests in submodules
+      run: |
+        make -C ctriface test
+        make -C ctriface test-man
+
+    - name: Run vHive end-to-end tests
```
Comment (ustiugov) [inline .github/workflows/tests.yml:61]: the following two jobs must separate from the subdirs job, right?

**Review by ustiugov — CHANGES_REQUESTED** [review]
please do add the cleaning script with the next commit.

File: function-images/lr_serving/Dockerfile (hunk 1 above)
Comment (shyamjesal) [reply function-images/lr_serving/Dockerfile:16]: [Scipy/fft](https://github.com/scipy/scipy/blob/v1.6.0/scipy/fft/_pocketfft/setup.py) build failed due to missing pybind11 dependency.

**Pull request closed without merging**